rclpy/src/rclpy/_rclpy_action.cpp

```diff
@@ -892,69 +163,12 @@ namespace rclpy
 void
 define_action_api(py::module m)
 {
-  m.def(
-    "rclpy_action_destroy_entity", &rclpy_action_destroy_entity,
-    "Destroy a rclpy_action entity.");
   m.def(
     "rclpy_action_get_rmw_qos_profile", &rclpy_action_get_rmw_qos_profile,
     "Get an action RMW QoS profile.");
-  m.def(
-    "rclpy_action_wait_set_add", &rclpy_action_wait_set_add,
-    "Add an action entitiy to a wait set.");
-  m.def(
-    "rclpy_action_wait_set_get_num_entities", &rclpy_action_wait_set_get_num_entities,
-    "Get the number of wait set entities for an action entitity.");
-  m.def(
-    "rclpy_action_wait_set_is_ready", &rclpy_action_wait_set_is_ready,
-    "Check if an action entity has any sub-entities ready in a wait set.");
-  m.def(
-    "rclpy_action_create_server", &rclpy_action_create_server,
-    "Create an action server.");
-  m.def(
-    "rclpy_action_server_is_available", &rclpy_action_server_is_available,
-    "Check if an action server is available for a given client.");
-  m.def(
-    "rclpy_action_take_goal_request", &rclpy_action_take_goal_request,
-    "Take a goal request.");
-  m.def(
-    "rclpy_action_send_goal_response", &rclpy_action_send_goal_response,
-    "Send a goal response.");
-  m.def(
-    "rclpy_action_take_result_request", &rclpy_action_take_result_request,
-    "Take a result request.");
-  m.def(
-    "rclpy_action_send_result_response", &rclpy_action_send_result_response,
-    "Send a result response.");
-  m.def(
-    "rclpy_action_take_cancel_request", &rclpy_action_take_cancel_request,
-    "Take a cancel request.");
-  m.def(
-    "rclpy_action_send_cancel_response", &rclpy_action_send_cancel_response,
-    "Send a cancel response.");
-  m.def(
-    "rclpy_action_publish_feedback", &rclpy_action_publish_feedback,
-    "Publish a feedback message.");
-  m.def(
-    "rclpy_action_publish_status", &rclpy_action_publish_status,
-    "Publish a status message.");
-  m.def(
-    "rclpy_action_accept_new_goal", &rclpy_action_accept_new_goal,
-    "Accept a new goal using an action server.");
-  m.def(
-    "rclpy_action_notify_goal_done", &rclpy_action_notify_goal_done,
-    "Notify and action server that a goal has reached a terminal state.");
-
   define_action_goal_handle(m);
+  define_action_server(m);
```

Comment:
I would put this in `rclpy_pybind11.cpp` with the goal of eventually deleting `_rclpy_action.cpp`.